pnpm workspace is not supported

### Describe the bug

Reproduction on Vitest `pnpm vitest --ui`

```
✔ Do you want to install @vitest/ui? … yes
 ERR_PNPM_ADDING_TO_ROOT  Running this command will add the dependency to the workspace root, which might not be what you want - if you really meant it, make it explicit by running this command again with the -w flag (or --workspace-root). If you don't want to see this warning anymore, you may set the ignore-workspace-root-check setting to true.
```

<img width="1022" alt="image" src="https://user-images.githubusercontent.com/6481596/190931730-24762573-6c10-4c29-9fee-0ce37c2db58a.png">


~~I will make a PR for it later.~~

### Reproduction

init a pnpm monorepo, and run `vitest --ui`

### System Info

```Shell
N/A
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/src/install.ts
+++ b/src/install.ts
@@ -1,3 +1,6 @@
+import { existsSync } from 'node:fs'
+import process from 'node:process'
+import { resolve } from 'node:path'
 import { execa } from 'execa'
 import { detectPackageManager } from '.'
 
@@ -27,6 +30,9 @@ export async function installPackage(names: string | string[], options: InstallP
     else
       args.unshift('--prefer-offline')
   }
+
+  if (agent === 'pnpm' && existsSync(resolve(options.cwd ?? process.cwd(), 'pnpm-workspace.yaml')))
+    args.unshift('-w')
 
   return execa(
     agent,

